code/src/components/Game.js

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import { NameInput } from "./NameInput";
+import { GameScreen } from "./GameScreen";
+import { WelcomeScreen } from "./WelcomeScreen";
+import { EndScreen } from "./EndScreen";
+import game from "../reducers/game";
+
+const Game = () => {
+  const userName = useSelector((store) => store.game.userName);
+  const gameStart = useSelector((store) => store.game.gameStart);
+  const gameOver = useSelector((store) => store.game.gameOver);
+  const gameCoords = useSelector((store) => store.game.gameData?.coordinates);
+
+  const dispatch = useDispatch();
+
+  if (gameCoords === "1,3") {
+    dispatch(game.actions.setGameOver(true));
```

Comment:
It is a nice solution, but it is not a future-proof one. What if the map structure will change on the backend (the game will grow for example)? However, in this case, we can safely assume that we are in the final room if there are **no remaining actions**

So our logic could be something like:

```jsx
if (actions.length === 0) {
   return <GameOver />;
}else{
...
```
or 
```jsx
...
actions.length === 0 && <EndScreen/>
...
```